src/lib.rs

```diff
@@ -0,0 +1,105 @@
+use std::collections::HashSet;
+
+use guest::prelude::*;
+use k8s_openapi::api::core::v1 as apicore;
+use kubewarden_policy_sdk::wapc_guest as guest;
+use lazy_static::lazy_static;
+extern crate kubewarden_policy_sdk as kubewarden;
+use kubewarden::{
+    logging, protocol_version_guest, request::ValidationRequest, response::ValidationResponse,
+    validate_settings,
+};
+use slog::{o, warn, Logger};
+
+use settings::Settings;
+mod settings;
+
+lazy_static! {
+    static ref LOG_DRAIN: Logger = Logger::root(
+        logging::KubewardenDrain::new(),
+        o!("policy" => "priority-class-policy")
+    );
+}
+
+#[no_mangle]
+pub extern "C" fn wapc_init() {
+    register_function("validate", validate);
+    register_function("validate_settings", validate_settings::<Settings>);
+    register_function("protocol_version", protocol_version_guest);
+}
+
+fn validate(payload: &[u8]) -> CallResult {
+    let validation_request: ValidationRequest<Settings> = ValidationRequest::new(payload)?;
+    match validation_request.extract_pod_spec_from_object() {
+        Ok(Some(pod_spec)) => {
+            match validate_pod_priority_class(
+                pod_spec,
+                validation_request.settings.allowed_priority_classes,
+            ) {
+                Ok(_) => kubewarden::accept_request(),
+                Err(err) => kubewarden::reject_request(Some(err.to_owned()), None, None, None),
+            }
+        }
+        Ok(None) => {
+            warn!(LOG_DRAIN, "no PodSpec found");
+            kubewarden::accept_request()
+        }
+        Err(err) => {
+            warn!(LOG_DRAIN, "cannot extract PodSpec");
+            Ok(serde_json::to_vec(&ValidationResponse {
+                accepted: true,
+                message: None,
+                code: None,
+                mutated_object: None,
+                audit_annotations: None,
+                warnings: Some(vec![format!(
+                    "Priority class policy failed to extract PodSpec from the request: {}",
+                    err.to_string()
+                )]),
+            })?)
```

Comment:
What I was proposing was to let the policy accept the incoming request, but add the error to the log message. Now, I don't remember the right syntax to be used by our logging library, but I wanted to have something like that:

```rust
err!(LOG_DRAIN, error = ?err, "cannot extract podSpec");
Ok(())
```

However, I think it's probably better to fail close and just reject. However, please change this code to make use of [`kubewarden::reject_request`](https://docs.rs/kubewarden-policy-sdk/latest/kubewarden_policy_sdk/fn.reject_request.html)
```